tests/test_offline_utils.py

```diff
@@ -0,0 +1,36 @@
+from io import BytesIO
+
+import pytest
+
+import requests
+from huggingface_hub.file_download import http_get
+
+from .testing_utils import (
+    OfflineSimulationMode,
+    RequestWouldHangIndefinitelyError,
+    offline,
+)
+
+
+def test_offline_with_timeout():
+    with offline(OfflineSimulationMode.CONNECTION_TIMES_OUT):
+        with pytest.raises(RequestWouldHangIndefinitelyError):
```

Comment:
maybe use `with self.assertRaises` to be consistent with other tests here @lhoestq (not a big deal if we want to stay consistent with other implems in our librairies)